store/tikv/txn.go

```diff
@@ -369,9 +368,7 @@ func (txn *KVTxn) Commit(ctx context.Context) error {
 	// pessimistic transaction should also bypass latch.
 	if txn.store.txnLatches == nil || txn.IsPessimistic() {
 		err = committer.execute(ctx)
-		if val == nil || sessionID > 0 {
```

Comment:
Will this  make the following logic executed unexpected for the inner session statements whose session id is 0?

Reply:
For internal queries, `val` is always nil; for external queries `sessionID > 0` always hold,
so `val == nil || sessionID > 0` is basically always `true`? @MyonKeminta

Reply:
Now I've change it to:
 `sessionID > 0 ` => onCommit
`sessionID == 0` => nothing

The original author means to call `onCommit` for external queries (although `val == nil` may be confusing in the old code).
@cfzjywxk

Reply:
maybe it's hard to distinguish external from internal queries form `sessionID`?

InRestrictedSQL also can call doCommit

https://github.com/pingcap/tidb/blob/c3f9b0899d2a5c7ab8da0e37f358596f2fe27913/session/session.go#L521

Reply:
If we take a session from the session pool to execute the SQL, `sessionID` is not set.
However, if we direct use the session to execute a internal SQL, that might be a problem.